halo/attest/voter/voter.go

```diff
@@ -506,33 +513,35 @@ func (v *Voter) instrumentUnsafe() {
 	count(v.proposed, proposedCount)
 }
 
-// detectReorg returns an error if the previous block doesn't match the new block's parent hash.
-// This indicates that a reorg occurred.
-func detectReorg(chainVer xchain.ChainVersion, chainVerName string, prevBlock xchain.Block, block xchain.Block, cursors map[xchain.StreamID]uint64) error {
-	if prevBlock.BlockHash == (common.Hash{}) {
-		return nil // Skip previous blocks without parent hash (init or consensus chain without block hashes).
+// detectReorg returns an error if a reorg is detected based on the following conditions:
+// - Previous block hash doesn't match the next block's parent hash.
+// - Stream offsets are not consecutive.
+func detectReorg(chainVer xchain.ChainVersion, prevBlock *xchain.Block, block xchain.Block, streamOffsets map[xchain.StreamID]uint64) error {
+	if prevBlock == nil {
+		return nil // Skip first block (without previous).
 	}
 
 	if prevBlock.BlockHeight+1 != block.BlockHeight {
 		return errors.New("consecutive block height mismatch [BUG]", "prev_height", prevBlock.BlockHeight, "new_height", block.BlockHeight)
 	}
 
 	for _, xmsg := range block.Msgs {
-		cursor, ok := cursors[xmsg.StreamID]
-		if ok && xmsg.StreamOffset != cursor+1 {
-			return errors.New("consecutive message offset mismatch", "streamID", xmsg.StreamID, "prev_offset", cursor, "new_offset", xmsg.StreamOffset)
+		offset, ok := streamOffsets[xmsg.StreamID]
+		if ok && xmsg.StreamOffset != offset+1 {
+			return errors.New("non-consecutive message offsets", "streamID", xmsg.StreamID, "prev_offset", offset, "new_offset", xmsg.StreamOffset)
 		}
 
 		// Update the cursor
-		cursors[xmsg.StreamID] = xmsg.StreamOffset
+		streamOffsets[xmsg.StreamID] = xmsg.StreamOffset
 	}
 
+	if block.BlockHash == (common.Hash{}) {
+		return nil // Skip consensus chain blocks without block hashes.
+	}
 	if prevBlock.BlockHash == block.ParentHash {
 		return nil // No reorg detected.
 	}
 
-	reorgTotal.WithLabelValues(chainVerName).Inc()
```

Comment:
this was only incremented for block hash reorgs, not stream offset or height issues.